Indicators/Vortex.cs

```diff
@@ -12,121 +13,133 @@
-    }
-
-    /// <summary>
-    /// Gets the positive vortex indicator.
-    /// </summary>
-    public IndicatorBase<IndicatorDataPoint> PositiveVortexIndicator { get; } = new Identity("PositiveVortex");
-
-    /// <summary>
-    /// Gets the negative vortex indicator.
-    /// </summary>
-    public IndicatorBase<IndicatorDataPoint> NegativeVortexIndicator { get; } = new Identity("NegativeVortex");
+            PlusVI = new FunctionalIndicator<IndicatorDataPoint>("PlusVI",
+                input => CalculateVortexIndex(_plusVMSum, _atr.Current.Value * _period),
+                isReady => _plusVM.Count >= _period && _atr.IsReady,
+                () => { _plusVM.Clear(); _plusVMSum = 0m; });
 
-    /// <summary>
-    /// Gets a value indicating whether this indicator is ready and fully initialized.
-    /// </summary>
-    public override bool IsReady => _window.Count == _period + 1;
+            MinusVI = new FunctionalIndicator<IndicatorDataPoint>("MinusVI",
+                input => CalculateVortexIndex(_minusVMSum, _atr.Current.Value * _period),
+                isReady => _minusVM.Count >= _period && _atr.IsReady,
+                () => { _minusVM.Clear(); _minusVMSum = 0m; });
+        }
 
-    /// <summary>
-    /// Adds two VortexIndicator instances together.
-    /// </summary>
-    /// <param name="left">The left VortexIndicator.</param>
-    /// <param name="right">The right VortexIndicator.</param>
-    /// <returns>A new VortexIndicator instance representing the sum of the two indicators.</returns>
-    public static VortexIndicator operator +(VortexIndicator left, VortexIndicator right)
-    {
-        return new VortexIndicator(left._period + right._period);
-    }
+        protected override decimal ComputeNextValue(IBaseDataBar input)
+        {
+            if (_previousInput != null)
+            {
+                var plusVMValue = Math.Abs(input.High - _previousInput.Low);
+                var minusVMValue = Math.Abs(input.Low - _previousInput.High);
+
+                if (_plusVM.Count == _period)
+                {
+                    _plusVMSum -= _plusVM[0];
+                    _plusVM.RemoveAt(0);
+                }
+                if (_minusVM.Count == _period)
+                {
+                    _minusVMSum -= _minusVM[0];
+                    _minusVM.RemoveAt(0);
+                }
+
+                _plusVM.Add(plusVMValue);
+                _minusVM.Add(minusVMValue);
+
+                _plusVMSum += plusVMValue;
+                _minusVMSum += minusVMValue;
+            }
+
+            _previousInput = input;
+            _atr.Update(input);
+
+            if (_plusVM.Count == _period && _minusVM.Count == _period && _atr.IsReady)
+            {
+                PlusVI.Update(input.EndTime, CalculateVortexIndex(_plusVMSum, _atr.Current.Value * _period));
+                MinusVI.Update(input.EndTime, CalculateVortexIndex(_minusVMSum, _atr.Current.Value * _period));
```

Comment:
- Here you are calculating the value for these indicators but they also are functional indicators and are doing another calculation in their "calculate" function.
- Also, I don't think Vortex calls for an ATR of period `period` to then be multiplied by `period` but actually an ATR of period 1 and then you use the sum of the ATR in said period.
    - You should take a look at our `IndicatorExtensions` and the `Sum` indicator: this could be achieved by doing something like `_atrSum = new Sum("name", period).Of(_atr);`